toolkit/tools/imagegen/configuration/disk.go

```diff
@@ -21,13 +22,72 @@ type Disk struct {
 	RawBinaries        []RawBinary        `json:"RawBinaries"`
 }
 
+// checkOverlappingPartitions checks that start and end positions of the defined partitions don't overlap.
+func checkOverlappingePartitions(disk *Disk) (err error) {
+	partIntervals := [][]uint64{}
+	//convert  partition entries to array of [start,end] locations
+	for _, part := range disk.Partitions {
+		partIntervals = append(partIntervals, []uint64{part.Start, part.End})
+	}
+	//sorting paritions by start position
+	sort.Slice(partIntervals, func(i, j int) bool {
+		return partIntervals[i][0] < partIntervals[j][0]
+	})
+	//confirm each partition ends before the next starts
+	for i := 0; i < len(partIntervals); i++ {
+		if i < len(partIntervals)-1 {
+			if partIntervals[i][1] > partIntervals[i+1][0] {
+				return fmt.Errorf("a [Partition] with an end location %d overlaps with a [Partition] with a start location %d", partIntervals[i][1], partIntervals[i+1][0])
+			}
+		}
+	}
+	return
+}
+
+// checkMaxSizeCorrectness checks that MaxSize is non-zero for cases in which it's used to clear disk space. This check
+// also confirms that the MaxSize defined is large enough to accomodate all partitions. No partition should have an
+// end position that exceeds the MaxSize
+func checkMaxSizeCorrectness(disk *Disk) (err error) {
+	const (
+		realDiskType = "path"
+	)
+	//MaxSize is not relevant if target disk is specified.
+	if disk.TargetDisk.Type != realDiskType {
+		if disk.MaxSize <= 0 && len(disk.Partitions) != 0 {
+			return fmt.Errorf("a configuration without a defined target disk must have a non-zero MaxSize")
```

Comment:
I'd also mention something about the `len(disk.Partitions) != 0` check. If I understand this check correctly, it is fine to not have a positive `MaxSize` if we don't define any partitions.